subprojects/platform-jvm/src/main/java/org/gradle/env/internal/DefaultBuildEnvironment.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.env.internal;
+
+import net.rubygrapefruit.platform.SystemInfo;
+import org.gradle.env.Architecture;
+import org.gradle.env.BuildEnvironment;
+import org.gradle.env.OperatingSystem;
+
+import javax.inject.Inject;
+
+public class DefaultBuildEnvironment implements BuildEnvironment {
+
+    private final SystemInfo systemInfo;
+
+    private final org.gradle.internal.os.OperatingSystem operatingSystem;
+
+    @Inject
+    public DefaultBuildEnvironment(SystemInfo systemInfo, org.gradle.internal.os.OperatingSystem operatingSystem) {
+        this.systemInfo = systemInfo;
+        this.operatingSystem = operatingSystem;
+    }
+
+    @Override
+    public OperatingSystem getOperatingSystem() {
+        if (operatingSystem.isWindows()) {
+            return OperatingSystem.WINDOWS;
+        } else if (operatingSystem.isMacOsX()) {
+            return OperatingSystem.MAC_OS;
+        } else if (operatingSystem.isLinux()) {
+            return OperatingSystem.LINUX;
+        }
+        return OperatingSystem.OTHER;
```

Comment:
Why not support all the ones Gradle can detect?

Reply:
I have not thought further than what is being used in the actual Adoptium repository. I do see `org.gradle.internal.os.OperatingSystem` now. Would it be ok to make that part of the public API?